src/marqo/tensor_search/tensor_search.py

```diff
@@ -2648,3 +2691,33 @@ def delete_documents(config: Config, index_name: str, doc_ids: List[str]):
             document_ids=doc_ids,
         )
     )
+
+def normalize_vector(embeddings: Union[List[List[float]], ndarray]) -> List[List[float]]:
+    """
+    Normalizes a list of vectors (embeddings) to have unit length.
+
+    Args:
+        embeddings (Union[List[List[float]], ndarray]): A list of vectors or a numpy ndarray of vectors to be normalized.
+
+    Returns:
+        List[List[float]]: A list of normalized vectors.
+    """
+
+    # Convert the input embeddings to a numpy array
+    if embeddings.__class__ == ndarray:
+        embeddings_array = embeddings
+    else:
+        embeddings_array = np.array(embeddings)
+
+    # Calculate the magnitude (Euclidean norm) of each vector along the last axis
+    magnitude = np.linalg.norm(embeddings_array, axis = -1, keepdims=True)
+
+    # Normalize each vector by dividing by its magnitude, handle zero magnitude case
+    if magnitude != 0:
+        embeddings_array = embeddings_array / magnitude
+    else:
+        raise core_exceptions.ZeroMagnitudeVectorError(f"Zero magnitude vector detected, cannot normalize.")
```

Comment:
Unlike custom vector field, which is provided by user, multimodal combo field embedding is calculated based on embeddings of different fields. If somehow the manitude is zero, there's no easy way for user to fix it. It might not be a good user experience if we just return this error to user. However, this might be too rare to be a real concern for now. @farshidz , what's your thought?

Reply:
Yeah feels like a 400 still

Reply:
Discussed offline with @farshidz , will handle multimodal combo fields in a follow up PR

Reply:
Jira: https://s2search.atlassian.net/browse/MOSD-138?atlOrigin=eyJpIjoiYzZkOTM0NzFhMTYwNDU3NDk5YjkwZDc1ZjYzYzhjNGMiLCJwIjoiaiJ9